app/views/unattended/provisioning_templates/provision/atomic_kickstart_default.erb

```diff
@@ -42,16 +42,51 @@ rootpw --iscrypted <%= root_pass %>
 
 reboot
 
+<% if @dynamic -%>
+# Partition table dynamic generation
+%pre --log=/tmp/install.pre.dynamic.log
+<%= @host.diskLayout %>
+<%= section_end -%>
+<% end -%>
+
+%pre --log=/tmp/install.pre.custom.log
+# Custom pre snippet generated by "<%= template_name + " custom pre" %>"
+<%= snippet_if_exists(template_name + " custom pre") %>
+<%= section_end -%>
+
+# copy %pre log files into chroot
+%post --nochroot
+cp -vf /tmp/*pre*log /tmp/sysimage/root/
+<%= section_end -%>
+
 %post
 <%= snippet 'redhat_register' %>
 rm -f /etc/ostree/remotes.d/*.conf
 <%= snippet('remote_execution_ssh_keys') %>
 <%= snippet 'efibootmgr_netboot' %>
-(
-# Report success back to Foreman
-<%= snippet 'built' %>
-) 2>&1 | tee /mnt/sysimage/root/install.post.log
+touch /tmp/foreman_built
+<%= section_end -%>
 
-exit 0
+%post --log=/root/install.post.custom.log
+# Custom post snippet generated by "<%= template_name + " custom post" %>"
+<%= snippet_if_exists(template_name + " custom post") %>
+<%= section_end -%>
+
+<%#
+The last post section halts Anaconda to prevent endless loop in case HTTP request fails
+%>
+<% if (is_fedora && os_major < 20) || (rhel_compatible && os_major < 7) -%>
+%post
```

Comment:
It looks like you're using multiple `%post` sections. Is that allowed? If so, I learned something new today :)

Reply:
Totally allowed and a good practice actually.